boot/zephyr/CMakeLists.txt

```diff
@@ -487,6 +487,10 @@ if(CONFIG_MCUBOOT_BOOT_BANNER)
   zephyr_sources(kernel/banner.c)
 endif()
 
+if(CONFIG_FIND_NEXT_SLOT_HOOKS)
+  zephyr_sources(hooks_find_next_slot.c)
+endif()
```

Comment:
need another Kconfig to use this file so that users can keep `CONFIG_FIND_NEXT_SLOT_HOOKS` enabled but disable the other one and replace with their own implementation